### DISCUSSION TITLE
Long Documents for Summarization -> LLM Zero-Shot Multi-Label Classification

### DISCUSSION THREAD

**Discussion by reversingentropy** [iryna-kondr/scikit-llm#74] [Q&A]
1)
How do I summarize an exceptionally long, book-sized document for summarization?
I want to create a summary and then use the LLM classifier on it. The problem is that the "summary of summaries" approach takes too much time.

2)
When working with a zero-shot multi-label classifier, are individual texts I want to classify treated as separate requests to the LLM API, or are multiple texts combined into one request? Specifically, when using list of lists , would all the text inputs be included in a single prompt, or would they remain separate? How do we manage text limits if they are all aggregated?

**Comment by OKUA1** [comment]
1) There is no way to make a summary of a very long document without chunking or compressing it in some way first, as there is a hard limit introduced by the LLM context length.

2) Each sample is passed in a separate prompt. The sample should be a string and cannot be a list.
